File: db/migrate/0001_create_authors.rb
# -*- coding: utf-8 -*-

class CreateAuthors < ActiveRecord::Migration
  def change
    create_table(:authors) do |t|
      ## Database authenticatable
      t.string :username,           null: false
      t.string :name,               null: false
      t.string :email,              null: false
      t.string :url
      t.string :encrypted_password, null: false, default: ""

      ## Recoverable
      t.string   :reset_password_token
      t.datetime :reset_password_sent_at

      ## Rememberable
      t.datetime :remember_created_at

      t.timestamps
    end

    add_index :authors, :username,             :unique => true
    add_index :authors, :email,                :unique => true
    add_index :authors, :reset_password_token, :unique => true
  end
end

# eof

File: priv/repo/migrations/0001_create_authors.exs
defmodule Burp.Repo.Migrations.CreateAuthors do
  use Ecto.Migration

  def change do
    create table(:authors) do
      add :username, :string, null: false
      add :email, :string, null: false
      add :url, :string
      add :encrypted_password, :string, null: false, default: ""

      timestamps()
    end

    create unique_index(:authors, [:username])
    create unique_index(:authors, [:email])
  end
end
